src/main/java/LinkedBinarySearchTree.java

```diff
@@ -413,6 +414,39 @@ public String toString() {
         return builder.toString();
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null) {
+            return false;
+        }
+        if (!(o instanceof BinaryTree)) {
+            return false;
+        }
+        BinaryTree<T> that = (BinaryTree<T>) o;
+        if (this.size() != that.size()) {
+            return false;
+        }
+        for (T element : this) {
+            if (this.count(element) != that.count(element)) {
+                return false;
+            }
+        }
```

Comment:
Not sure I agree with bag equality here. My gut says structure should be part of a tree's equality. This is a little annoying to do though, since `size` is a property of the tree instead of the nodes, so you can't just do `this.root.equals(that.root) && Tree(this.left).equals(Tree(that.left)) && Tree(this.right).equals(Tree(that.right))`

Reply:
The public methods do have different behaviours when the structure is different. For example, `preOrderIterator`.